### PR TITLE
chore(eslint): fix non-destructive eslint errors

### PR DESCRIPTION
**Pull request by seaerchin**
Part 3 of #819 

## Problem
This is a follow up to #819 and focuses on eslint _chores_, which are code tidiness issues (eg: missing import/unused variables etc) rather than refactoring focused. 

## Solution
1. follow eslint
2. delete/add stuff as appropriate

## Notes
there might be some overlap with the previous part so please still keep your eyes peeled but the majority of code changes here should be relatively easy to review

### CODE REVIEW

File: src/components/Footer.jsx [hunk 1]
```diff
@@ -44,11 +43,4 @@ const Footer = ({
   </div>
 )
 
-Footer.propTypes = {
-  isDeleteDisabled: PropTypes.bool,
-  isSaveDisabled: PropTypes.bool,
-  deleteCallback: PropTypes.func,
-  saveCallback: PropTypes.func,
-}
-
```
Comment (alexanderleegs) [inline src/components/Footer.jsx:53]: Could I check why the proptypes here are removed?

File: src/components/GenericWarningModal.jsx [hunk 2]
```diff
@@ -54,7 +54,6 @@ const GenericWarningModal = ({
 GenericWarningModal.propTypes = {
   displayTitle: PropTypes.string.isRequired,
   displayText: PropTypes.string.isRequired,
-  displayImage: PropTypes.string,
```
Comment (alexanderleegs) [inline src/components/GenericWarningModal.jsx:57]: Same as above

**Review by alexanderleegs — APPROVED** [review]
minor nits, but otherwise lgtm

File: src/components/Footer.jsx (hunk 1 above)
Comment (seaerchin) [reply src/components/Footer.jsx:53]: the proptypes are never used as these don't exist as props @__@

File: src/components/GenericWarningModal.jsx (hunk 2 above)
Comment (seaerchin) [reply src/components/GenericWarningModal.jsx:57]: i removed it as it's unused - there's a similar `displayImg` prop up above though so actually i'll add this `PropType` back in + change the `displayImg` to `displayImage`

**Pull request merged**